pkg/virt-controller/watch/application.go

```diff
@@ -162,8 +159,12 @@ func Execute() {
 
 	app.rsInformer = app.informerFactory.VMIReplicaSet()
 
-	app.configMapInformer = app.informerFactory.ConfigMap()
-	app.configMapCache = app.configMapInformer.GetStore()
+	configMapInformer := app.informerFactory.ConfigMap()
+	stopChan := make(chan struct{}, 1)
+	defer close(stopChan)
+	go configMapInformer.Run(stopChan)
```

Comment:
umm. i'm not sure about starting the informer here like this.

The problem is that this informer came from the app,InformerFactory.  Later on when app.informerFactory.Start() is called, it's going to launch this same informer in yet another go routine.

You can call app.informerFactor.Start() multiple times safely though.

If you make this configMapInformer the first informer you create, and then immeditealy call informerFactor.Start(), only that informer will start because it's the only one that's been allocated. Later on when informerFactory.Start() is called again, only the informers that have been created since then get started. 

hopefully that makes sense. I know this is a bit confusing.

Reply:
I wasn't sure about this part - thanks for the explanation! I think it should be ok now